### DISCUSSION TITLE
Questions about Access Control

### DISCUSSION THREAD

**Discussion by hgjo-nanum** [prisma/prisma#23859] [Q&A]
### Question

hello.
I don't know if I can post this here, but I'm posting it because I don't know what to do.
I'm really sorry if I shouldn't post here like this.

I work at a software development company.
We are developing using next.js and considering ORM application.
We are actively considering prisma.
I needed access control, so I looked into it and it seems there are several ways to do it.
But among many methods, we would like to apply it to prisma itself.
So, I looked at the policy and found that it does not yet provide access control functionality. It says coming soon.
So I'm curious when it will be released and if you can provide a rough outline.

If this post is problematic, I will delete it.
Please let me know how I can reach you (e.g. email).

thank you

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment]
Hi @hgjo4824 👋 

Your question is absolutely welcome here and we are always ready to help.

Prisma does not directly provide access control functionality. However, you can implement authentication and authorization in your application using Prisma along with other tools. There are a few ways to approach it:

1. External Authorization Service: You can leverage a dedicated authorization service like Auth0, Firebase Authentication, or AWS IAM. These services handle user roles and permissions, and Prisma can integrate with them to enforce access rules on your data.

2. Application-Level Logic: Implement access control logic directly within your Next.js application. You can check user roles retrieved from an external service or stored locally (JWT tokens) before performing Prisma operations. This approach requires more development effort but offers greater control.

> I looked at the policy and found that it does not yet provide access control functionality. It says coming soon.

Can you please share the link to the policy you are referring to?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
